lib/engine/game/g_1880/game.rb

```diff
@@ -167,6 +177,73 @@ class Game < Game::Base
                   },
                   { name: '10', distance: 10, price: 1000, num: 10 }].freeze
 
+        EVENTS_TEXT = Base::EVENTS_TEXT.merge(
+          'float_30' => ['30% to Float', 'Corporations must have 30% of their shares sold to float'],
+          'float_40' => ['40% to Float', 'Corporations must have 40% of their shares sold to float'],
+          'float_60' => ['60% to Float', 'Corporations must have 60% of their shares sold to float'],
+          'permit_b' => ['B Permit', 'Only corporations with a B building permit can build track'],
+          'permit_c' => ['C Permit', 'Only corporations with a C building permit can build track'],
+          'permit_d' => ['D Permit', 'Only corporations with a D building permit can build track'],
+        ).freeze
+
+        def event_float_30!
+          @log << "-- Event: #{EVENTS_TEXT['float_30'][1]} --"
+          update_float_percent(30)
+        end
+
+        def event_float_40!
+          @log << "-- Event: #{EVENTS_TEXT['float_40'][1]} --"
+          update_float_percent(40)
+        end
+
+        def event_float_60!
+          @log << "-- Event: #{EVENTS_TEXT['float_60'][1]} --"
+          update_float_percent(60)
+        end
+
+        def update_float_percent(percent)
+          @corporations.each do |c|
+            next if c.type == :minor || c.floated?
+
+            c.float_percent = percent
+          end
+        end
+
+        def event_permit_b!
+          @log << "-- Event: #{EVENTS_TEXT['permit_b'][1]} --"
+          @active_building_permit = 'B'
+        end
+
+        def event_permit_c!
+          @log << "-- Event: #{EVENTS_TEXT['permit_c'][1]} --"
+          @active_building_permit = 'C'
+        end
+
+        def event_permit_d!
+          @log << "-- Event: #{EVENTS_TEXT['permit_d'][1]} --"
+          @active_building_permit = 'D'
+        end
+
+        def setup
+          @foreign_investors_can_build = true
+          setup_building_permits
+          setup_unsaleable_shares
+        end
+
+        def setup_building_permits
+          @corporation_building_permits = {}
+          @active_building_permit = 'A'
+          @building_permit_choices_by_president_percent = {
+            20 => %w[ABC BCD],
+            30 => %w[AB BC CD],
+            40 => %w[A B C D],
+          }
+        end
+
+        def setup_unsaleable_shares
```

Comment:
did you merge this with the setup method in my PR?

https://github.com/tobymao/18xx/blob/f3ee29ccd2bdaf82f7881e1ddad146012ecc095a/lib/engine/game/g_1880/game.rb#L259

Reply:
No, I'll merge our PRs once your PR is merged into master